eos-chrome-plugin-update-upstream

```diff
@@ -21,6 +21,21 @@ DEST_DIR=/var/lib/chromium-plugins-extra/eos-chrome-plugin-updater
 WGET_COMMAND="wget -c --tries=5"
 DAYS_CHECK=7
 
+# Data needed for the Flash plugin, from Adobe
+FLASH_FALLBACK_VERSION="23.0.0.185"
+FLASH_FALLBACK_URL="https://fpdownload.adobe.com/pub/flashplayer/pdc/${FLASH_FALLBACK_VERSION}/flash_player_ppapi_linux.$(arch).tar.gz"
+FLASH_FALLBACK_SHA256SUM="3f7a13a615d3cf19e5feb02485e12e3785a6ef07cd7ae32072cd860468fd322b"
+
+FLASH_LKGV_FILE_URL="https://raw.githubusercontent.com/endlessm/eos-chrome-plugin-updater/master/flashplugin-$(arch).lkgv"
```

Comment:
> I'm not sure about having production code pulling from a master branch. I'm also not sure about having all users start by trying to read a file from github, though I suppose we have the fallback. Probably worth getting the opinion of @cosimoc.

I'm not sure either, but this was the only place I could hang it from so far :) If somebody could get the file hang out of the S3 bucket and give me the URL, I'll add it to the script.

> How secure is it to reach back to github to get the URL and SHA256SUM? Wouldn't that be open to a simple attack of injecting an arbitrary URL with the associated SHA256SUM? 

I think that's exactly as secure as when you download a big file and its associated checksum from the same server :-): you can check the integrity of the downloaded file is intact (which is what I'm mostly interested in here), but you can't protect against a MITM attack, of course. Anway I'd keep this, it's better than nothing.

> If not really secure, why not simply download the latest from Adobe rather than specify a version? Or, is there no way to do this without calling out a specific version?

I could not find a straightforward way of getting the last version from Adobe, as there's no global index or the like you can check, all you have is https://get.adobe.com/flashplayer/otherversions, where you click on combo boxes until a "Download" button lights up and takes you to a URL that redirects into the final file.

Actually we used to download hardcoded versions in the past too (see https://github.com/endlessm/eos-flashplugin-updater/blob/master/eos-flash-update#L7), probably for the same reason?

> I am not sure I fully understand the logic behind the lkgv file. Is it so that we can update the plugin version that is going to be downloaded without pushing out an OS update?

Yes, that's the idea

> Pulling from github directly may be OK for now, but we probably want to host these files on S3 later on. We may also need to have different URLs depending on which branch/OS release we're on...
However, is there no APT index file on the Adobe servers we can piggyback on?

No, there's nothing like that.

However, I found that if you request this URL: https://get.adobe.com/flashplayer/webservices/json/?platform_type=Linux&platform_arch=x86-64&eventname=flashplayerotherversions

...you get a JSON file with and array of three elements, where the first one seems to contain the information relevant to the PPAPI plugin we need: Version, URL... and so forth.

The problem with that is that, and besides the fact that such URL might stop working or change its behaviour at any point, it seems to be like a bit overkill to me to do that parsing from the script itself so I still think we should rely on the LKGV file instead (but placed in a better location, probably in a S3 bucket, as you said).

What we could do, though, is to have a Jenkins job that would run some other script that would periodically check that URL, parse its contents and eventually update the LKGV file. How does that sound?